src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1453,7 +1529,11 @@ public void Process_StartTest()
             Assert.Equal(userName, p.StartInfo.UserName);
             Assert.Same(password, p.StartInfo.Password);
             Assert.Equal(domain, p.StartInfo.Domain);
+
+            // Make sure that process is fully created before killing it to avoid AccessDeniedExceptions.
+            Thread.Sleep(500);
```

Comment:
Instead of thread.sleep, why dont you check if p.Id is > 0. If process hasn't started yet, this will throw InvalidOperationException.